File: 1/two-sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int, len(nums))
    for i, num := range nums {
        index, ok := m[target - num]
        if ok {
            return []int{index, i}
        }
        m[num] = i
    }
    return []int{}
}

File: 1/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const indices = new Map();
    for (let index = 0; index < nums.length; index++) {
        const currentNum = nums[index];
        const complement = target - currentNum;
        if (indices.has(complement)) {
            return [indices.get(complement), index];
        }
        indices.set(currentNum, index);
    }
    return [];
};
